### ISSUE TITLE
Agent view page needs to be more resilient with missing Datasets

### ISSUE THREAD

**Issue by weyrick** [ns1labs/orb#948]
Right now the Agent View page breaks if a Dataset reported by the agent heartbeat is not found. The UI assumes that any dataset id reported by the agent should be able to be looked up (at /policies/dataset/:id:) successfully, and when it is not found reports a 404 and does not show the View page.

The problem is that it takes time for Dataset deletes to propagate to the agent and back to through the heartbeat. So once a Dataset is deleted, the agent has to receive the notification of delete, adjust the backend, and send a heartbeat back with the deleted dataset removed. During this time window, the Agent View page will break because the latest agent heartbeat will still reflect the deleted Dataset.

The solution is for the UI to handle missing datasets more gracefully - it cannot assume that a dataset ID reported by the agent actually exists in the control plane. If it does not exist, the Agent View page should still continue to load, and it should indicate to the user that there is a missing Dataset.

**Comment by manrodrigues** [comment]
@gpazuch 

We need to find a way to deal with the missing dataset in the minute before the next heartbeat (which will update the data)


Look at this scenario:

- We have two policies applied to the agent, each through a different dataset
- When we remove only one dataset, in the next minute a missing dataset span will be shown and no policy or dataset will be shown in the agent (nor the one that remains)
- Only after one minute the policies and datasets still linked to the agent will be shown again
This can give the wrong impression that everything has been removed.

![failed_dataset-min](https://user-images.githubusercontent.com/78241475/159739168-310d9eae-a57d-4922-ba7d-228dbaa16f7c.gif)

**Comment by manrodrigues** [comment]
I guess this was not solved. Actually, this is failing now for groups and datasets removal:

groups:

![remove_group_failed](https://user-images.githubusercontent.com/78241475/161070400-212ed273-1882-4602-b995-3b17e24d3e00.gif)

datasets:

![remove_dataset_failed](https://user-images.githubusercontent.com/78241475/161070462-14474df7-5d60-454f-b124-a5c2acb59da1.gif)

Since no new PR was linked, I think this card is here in QA improperly. @iatdaitan @gpazuch @dscabral can you check it, please?

**Comment by gpazuch** [comment]
@manrodrigues I think I can improve the ui messages a bit, but not much I can do about the missing/deleted entities, this agent contains invalid references.

**Comment by gpazuch** [comment]
https://github.com/ns1labs/orb/pull/1017

**Comment by manrodrigues** [comment]
ok

![image](https://user-images.githubusercontent.com/78241475/162481692-0d329ff3-35d4-4b05-93c4-4e5a43e6b2d5.png)

**Comment by manrodrigues** [comment]
@gpazuch why the message is "dataset policy"? what you mean?

**Comment by gpazuch** [comment]
> @gpazuch why the message is "dataset policy"? what you mean?

@manrodrigues I'm unsure of what you mean by that.

**Comment by gpazuch** [comment]
related to #1069

**Comment by manrodrigues** [comment]
this is not working yet.. 

1 - the span message  is still "dataset policy"

![image](https://user-images.githubusercontent.com/78241475/164601294-cfa12a77-6c27-4d85-a046-debca5ecd7df.png)

2 - when a policy is applied through more than one dataset, even when only one dataset is removed from orb, the policy stop to be displayed on agent view's page while the ui failed to fetch the dataset removed:

![no_policies](https://user-images.githubusercontent.com/78241475/164602521-9f472b2d-b167-4664-9019-1430b3df6971.gif)

**Comment by manrodrigues** [comment]
@gpazuch it's related with some new merged PR? Should I test it again?

**Comment by manrodrigues** [comment]
@gpazuch 

1- The groups linked are not being displayed anymore:

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/78241475/183478446-9b98582f-03d2-490b-bce3-6cdf9d38ff62.png)

2 - When more than 1 dataset is applied to the same policy, the page try to GET a request with multiple id's comma separated.. it cause an intrnal server error (500)

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/78241475/183478473-c0c19a1d-9224-4b09-8d01-39dc45be5102.png)

3 - If 1 dataset could not be displayed, no other dataset is being displayed

**Comment by manrodrigues** [comment]
ok

![image](https://user-images.githubusercontent.com/78241475/183704188-ed23681f-4e85-4607-9905-3428a4f7d8b6.png)

![missing_dataset](https://user-images.githubusercontent.com/78241475/183704448-861da041-48a9-4c9d-acc0-625e4e9cd9af.gif)
